core/app/models/concerns/spree/user_reporting.rb

```diff
@@ -1,22 +1,35 @@
 module Spree
   module UserReporting
-    extend DisplayMoney
-    money_methods :lifetime_value, :average_order_value
+    def display_lifetime_value(store: Store.default, currency: store.default_curremcy)
+      Spree::Money.new(lifetime_value(store: store, currency: currency), currency: currency)
+    end
+
+    def lifetime_value(**args)
+      order_calculate(**args,
+                      operation: :sum,
+                      column: :total)
+    end
+
+    def display_average_order_value(store: Store.default, currency: store.default_currency)
+      Spree::Money.new(average_order_value(store: store, currency: currency), currency: currency)
+    end
 
-    def lifetime_value
-      orders.complete.sum(:total)
+    def average_order_value(**args)
+      order_calculate(**args,
+                      operation: :average,
+                      column: :total)
     end
 
-    def order_count
-      orders.complete.size
+    def order_count(store = nil)
+      store ||= Store.default
+      order_calculate(store: store,
+                      currency: store.supported_currencies.split(','),
+                      operation: :count,
+                      column: :all)
     end
 
-    def average_order_value
-      if order_count > 0
-        lifetime_value / order_count
-      else
-        BigDecimal('0.00')
-      end
+    def order_calculate(store:, currency:, operation:, column:)
```

Comment:
@tomurb let's make this method private as this is our internal API and we should not expose it